singularity_compatibility.rst

```diff
@@ -0,0 +1,149 @@
+applications and data with {Project} without concerns of image format
+incompatibility when running in a different computing environment or
+collaborating with colleagues that still use a Singularity installation.
+
+
+.. _singularity_environment_variable_compatibility:
+
+***************************************************
+ Singularity Prefixed Environment Variable Support
+***************************************************
+
+{Project} respects environment variables with the ``SINGULARITY_`` and
+``SINGULARITYENV_`` prefixes when their respective ``{ENVPREFIX}_`` and
+``{ENVPREFIX}ENV_`` counterparts are not set.
+
+
+This first command does not use any environment variable compatibility behavior
+as it is using the ``{ENVPREFIX}ENV_`` prefix to command {Project} to create an
+environment variable in the container with the name ``FOO`` and the value
+``BAR``:
+
+.. code::
+
+  $ APPTAINERENV_FOO=BAR apptainer exec docker://alpine env
+  [...]
+  FOO=BAR
+
+We can see from the ``env`` output that this environment variable was properly
+set inside the container.
+
+Next we can use the ``SINGULARITYENV_`` prefix to do the same thing, but this
+time we wil have the intended value be ``BAZ``:
+
+  $ SINGULARITYENV_FOO=BAZ apptainer exec  docker://alpine env
+  WARNING: DEPRECATED USAGE: Forwarding SINGULARITYENV_FOO as environment variable will not be supported in the future, use APPTAINERENV_FOO instead
+  [...]
+  FOO=BAZ
+
+Notice that we have a warning for ``DEPRECATED USAGE`` when doing so. This is
+because future version of {Project} may stop supporting environment variable
+compatibility once it is past this current period of transition.
+
+Finally, if both are set, the value set by the ``{ENVPREFIX}ENV_`` variable will
+take priority over the ``SINGULARITYENV_`` variable.
+
+.. code::
+
+  $ APPTAINERENV_FOO=BAR SINGULARITYENV_FOO=BAZ apptainer exec iqube_latest.sif env
+  WARNING: Skipping environment variable [SINGULARITYENV_FOO=BAZ], FOO is already overridden with different value [BAR]
+  [...]
+  FOO=BAR
+
+In this case a warning is omitted to let the user know that two variables were
+set to create the same environment variable in the container in case they were
+unaware of one of them existing in their environment.
+
+
+*****************************
+ Singularity Command Symlink
+*****************************
+
+With the same intention as the environment variable handling, {Project}
+installations will include a symlink to the ``{command}`` binary named
+``singularity``. This will allow existing tooling and scripts using the
+``singularity`` command to continue to operate after a migration to {Project}
+has taken place. Below is an example of running the ``version`` command using
+either program name and getting the same output because we are running the same
+underlying binary:
+
+.. code::
+
+  $ apptainer version
+  {InstallationVersion}
+  $ singularity version
+  {InstallationVersion}
```

Comment:
If you use `--version` here instead of `version` it will add `{command} version` as a prefix on the output of each and make it clearer that the underlying program is the same.